libpromises/pipes_unix.c

```diff
@@ -142,6 +169,113 @@ static pid_t CreatePipeAndFork(const char *type, int *pd)
 
 /*****************************************************************************/
 
+static pid_t CreatePipeAndFork(const char *type, int *pd)
+{
+    IOPipe pipes[2] = {{0}};
+    pipes[0].type = type;
+    
+    pid_t pid = GenericCreatePipeAndFork(pipes);
+    
+    pd[0] = pipes[0].pipe_desc[0];
+    pd[1] = pipes[0].pipe_desc[1];
+
+    return pid;
+}
+
+/*****************************************************************************/
+
+static pid_t CreatePipesAndFork(const char *type, int *pd, int *pdb)
+{
+    IOPipe pipes[2] = {{0}};
```

Comment:
Indeed. In fact I don't have to initialize this at all. If some descriptor won't be valid after this function exits return value will indicate error.